consensus/src/extension/mod.rs

```diff
@@ -34,14 +38,8 @@ impl<H: Hasher, D: Data, MK: MultiKeychain, FH: FinalizationHandler<D>> Ordering
     }
 
     fn handle_batch(&mut self, batch: Vec<DagUnit<H, D, MK>>) {
-        for unit in batch {
-            let unit = unit.unpack();
-            self.finalization_handler.unit_finalized(
-                unit.creator(),
-                unit.round(),
-                unit.as_signable().data().clone(),
-            )
-        }
+        let batch = batch.into_iter().map(|unit| unit.into()).collect();
```

Comment:
An additional (minor) advantage of having a separate trait is that you could take a `Vec` of something that's `Into` instead of doing this here.

Reply:
changed